source/lex.tex

```diff
@@ -1327,14 +1327,15 @@
 cannot be represented by any type in its list and
 an extended integer type\iref{basic.fundamental} can represent its value,
 it may have that extended integer type.
-If all of the types in the list for the \grammarterm{integer-literal}
-are signed,
-the extended integer type is signed.
-If all of the types in the list for the \grammarterm{integer-literal}
-are unsigned,
-the extended integer type is unsigned.
-If the list contains both signed and unsigned types,
-the extended integer type may be signed or unsigned.
+This extended integer type shall have the following signedness:
+\begin{itemize}
+\item If all types in the list for the \grammarterm{integer-literal}
+are signed, it is signed.
+\item If all types in the list for the \grammarterm{integer-literal}
+are unsigned, it is unsigned.
+\item If the list contains both signed and unsigned types,
+it may be signed or unsigned.
```

Comment:
In order to make the standard as readable as possible, I feel that the antecedent for "it" should be the immediately preceding noun that matches grammar-wise. In this last bullet, it seems that "it" refers to the list. (Those in the know understand that a list is not supposed to be signed or unsigned, but the point of the standard is to convey knowledge, not to assume the reader has knowledge.)

(This is obviously violated in the status quo quite a bit, so this is more an aspirational comment for new edits.)

If you want to avoid the repetition, introduce "T" for the type we're talking about.

Given the editorial overlap with CWG/issues#404, I'm wondering whether the bulletization (and possibly slightly larger massagings) should be left to the CWG issue.

Reply:
> If you want to avoid the repetition, introduce "T" for the type we're talking about.

Sounds like a good plan, I've done that.

> Given the editorial overlap with CWG/issues#404, I'm wondering whether the bulletization (and possibly slightly larger massagings) should be left to the CWG issue.

I've thought about it too, and don't think it's strictly necessary. The CWG issue is just about putting *implementation-defined* somewhere in the wording. The diff suggested in the core issue makes sense before and after the editorial changes here, so the two are somewhat independent.